packages/gatsby/src/db/sanitize-node.js

```diff
@@ -3,19 +3,25 @@ const _ = require(`lodash`)
 /**
  * Make data serializable
  * @param {(Object|Array)} data to sanitize
+ * @param {boolean} isNode = true
+ * @param {Set<string>} path = new Set
  */
-const sanitizeNode = (data, isNode = true) => {
+const sanitizeNode = (data, isNode = true, path = new Set()) => {
   const isPlainObject = _.isPlainObject(data)
 
+  path.add(data)
```

Comment:
Should this be inside `if (isPlainObject || _.isArray(data))` block (where we want to traverse things)? If it's not we might add primitives here and wrongly omit some fields as we are traversing because of that?

Reply:
That's a good point! I'll change that.